### PR TITLE
Authorization rule

### PR DESCRIPTION
**Pull request by SiansandaTamara**
## The fokkowing were implemented ##
- Install CanCanCan in your project.
- Add a role column to the users table. Remember to use a migration for this.
- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
- For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
- For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+      return unless user.present? # additional permissions for logged in users (they can read their own posts)
+
+      can :read, Post, user: user
+
+      return unless user.admin? # additional permissions for administrators
+
+      can :read, Post
+    end
+  end
```
Comment (Mihndim2020) [inline app/models/ability.rb:40]: - [ ] As per the project [requirments](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_add_authorization.md);

- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). 
- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). 
Please kindly implement the above requirement, something like so:

 class Ability
 include CanCan::Ability

     def initialize(user)

         user ||= User.new # guest user (not logged in)

         if user.admin? 
         can :manage, :all
         else
         can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
         can :read, :all
         end
     end
 end

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -27,6 +27,20 @@ def create
     end
   end
```
Comment (Mihndim2020) [inline app/controllers/posts_controller.rb:27]: - [ ] Kindly use **CanCanCan** for Post authorization in `Post Controller`, something like so:

`load_and_authorize_resource`

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @SiansandaTamara, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Clean/Organized code
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShahierNashaat — APPROVED** [review]
Hi  @SiansandaTamara, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Good overall design ✔️
- Your code is clean and well documented ✔️

# Approved :ok_hand: :ok_hand: :ok_hand:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**